src/cpp/src/whisper_pipeline.cpp

```diff
@@ -28,21 +29,52 @@ ov::genai::OptionalWhisperGenerationConfig get_config_from_map(const ov::AnyMap&
     }
 }
 
-ov::genai::ChunkStreamerVariant get_chunk_streamer_from_map(const ov::AnyMap& config_map) {
-    ov::genai::ChunkStreamerVariant streamer = std::monostate();
-
-    if (config_map.count(ov::genai::utils::STREAMER_ARG_NAME)) {
-        auto any_val = config_map.at(ov::genai::utils::STREAMER_ARG_NAME);
-        if (any_val.is<std::shared_ptr<ov::genai::ChunkStreamerBase>>()) {
-            streamer = any_val.as<std::shared_ptr<ov::genai::ChunkStreamerBase>>();
-        } else if (any_val.is<std::function<bool(std::string)>>()) {
-            streamer = any_val.as<std::function<bool(std::string)>>();
-        } else if (any_val.is<std::function<ov::genai::StreamingStatus(std::string)>>()) {
-            streamer = any_val.as<std::function<ov::genai::StreamingStatus(std::string)>>();
-        }
+ov::genai::StreamerVariant get_streamer_from_map(const ov::AnyMap& config_map) {
+    ov::genai::StreamerVariant streamer = std::monostate();
+
+    if (!config_map.count(ov::genai::utils::STREAMER_ARG_NAME)) {
+        return streamer;
+    }
+
+    auto any_val = config_map.at(ov::genai::utils::STREAMER_ARG_NAME);
+
+    OPENVINO_SUPPRESS_DEPRECATED_START
+    if (any_val.is<std::shared_ptr<ov::genai::StreamerBase>>()) {
+        streamer = any_val.as<std::shared_ptr<ov::genai::StreamerBase>>();
+    } else if (any_val.is<std::function<bool(std::string)>>()) {
+        streamer = any_val.as<std::function<bool(std::string)>>();
+    } else if (any_val.is<std::function<ov::genai::StreamingStatus(std::string)>>()) {
+        streamer = any_val.as<std::function<ov::genai::StreamingStatus(std::string)>>();
     }
+    OPENVINO_SUPPRESS_DEPRECATED_END
+
     return streamer;
 }
+
+std::shared_ptr<ov::genai::StreamerBase> to_base_streamer(ov::genai::StreamerVariant streamer,
+                                                          ov::genai::Tokenizer& tokenizer) {
+    std::shared_ptr<ov::genai::StreamerBase> streamer_ptr;
+
+    OPENVINO_SUPPRESS_DEPRECATED_START
```

Comment:
should we suppress only `ChunkStreamerBase` related code?

Reply:
done